File: Первый курс/Первый семестр/Практикум на ЭВМ/Паскаль/7.pas
var
  a: array[1..256, 1..256] of integer;
  n, m, i, j, k, k_min, i_max, j_max, t: integer;

begin
  repeat
    write('Enter n, m: ');
    readln(n, m);
  until (n in [1..256]) and (m in [1..256]);
  for i := 1 to n do
  begin
    for j := 1 to m do
    begin
      readln(a[i, j]);
    end;
    writeln();
  end;
  writeln('Matrix:');
  for i := 1 to n do
  begin
    for j := 1 to m do
    begin
      write(a[i, j], #9);
    end;
    writeln;
  end;
  i_max := 1; j_max := 1;
  for i := 1 to n do
    for j := 1 to m do
      if  abs(a[i, j]) > abs(a[i_max, j_max]) then
      begin
        i_max := i;
        j_max := j;
      end;
  writeln('Max = a[', i_max, ',', j_max, '] = ', a[i_max, j_max]);
  if n < m then k_min := n
  else k_min := m;
  repeat
    write('Enter k ');
    readln(k);
  until k in [1..k_min];
  for i := 1 to n do
  begin
    t := a[i, k];
    a[i, k] := a[i, j_max];
    a[i, j_max] := t;
  end;
  for j := 1 to m do
  begin
    t := a[k, j];
    a[k, j] := a[i_max, j];
    a[i_max, j] := t;
  end;
  writeln('New Matrix:');
  for i := 1 to n do
  begin
    for j := 1 to m do
      write(a[i, j], #9);
    writeln();
  end;
end.

File: Первый курс/Первый семестр/Практикум на ЭВМ/C++/7.cpp
#include <cmath>
#include <iostream>
using namespace std;
int main() {
  int n, m;
  std::cout << "Enter n,m" << '\n';
  cin >> n >> m;
  int a[n][m];
  for (size_t i = 0; i < n; i++) {
    for (size_t j = 0; j < m; j++) {
      cin >> a[i][j];
    }
  }
  cout << "Matrix:" << '\n';
  for (size_t i = 0; i < n; i++) {
    for (size_t j = 0; j < m; j++) {
      cout << a[i][j] << '\t';
    }
    cout << '\n';
  }
  int i_max = 0, j_max = 0;
  for (size_t i = 0; i < n; i++) {
    for (size_t j = 0; j < m; j++) {
      if (abs(a[i][j]) > abs(a[i_max][j_max])) {
        i_max = i;
        j_max = j;
      }
    }
  }
  cout << "Max a[" << i_max + 1 << "][" << j_max + 1 << "]"
       << "=" << a[i_max][j_max] << '\n';
  int k;
  cout << "Enter k" << '\n';
  cin >> k;
  k--;
  int t;
  for (size_t i = 0; i < n; i++) {
    t = a[i][k];
    a[i][k] = a[i][j_max];
    a[i][j_max] = t;
  }
  for (size_t j = 0; j < m; j++) {
    t = a[k][j];
    a[k][j] = a[i_max][j];
    a[i_max][j] = t;
  }
  cout << "New Matrix" << '\n';
  for (size_t i = 0; i < n; i++) {
    for (size_t j = 0; j < m; j++) {
      cout << a[i][j] << '\t';
    }
    cout << '\n';
  }
  return 0;
}
